### PR TITLE
MTA-2812 New "Personal details" page

### PR DESCRIPTION
**Pull request by richardjamesbaconHMRC**

### CODE REVIEW

File: app/views/cards/home/nationalInsurance.scala.html [hunk 1]
```diff
@@ -13,4 +13,4 @@ <h3 class="heading-small no-margin-top">@Messages("label.national_insurance")</h
       <li><a class="ga-track-anchor-click" href="/check-your-state-pension/account/nirecord" data-ga-event-category="link - click" data-ga-event-action="Income" data-ga-event-label="View gaps in your record">@Messages("label.view_gaps_in_your_record")</a></li>
     </ul>
   </div>
-</div>
+</div>
```
Comment (nicf82) [inline app/views/cards/home/nationalInsurance.scala.html]: Add newline again

Comment (richardjamesbaconHMRC) [reply app/views/cards/home/nationalInsurance.scala.html]: Done

File: app/util/DateTimeTools.scala [hunk 2]
```diff
@@ -47,4 +49,15 @@ object DateTimeTools {
 
   def short(dateTime: DateTime) = formatter("dd/MM/yyy").print(dateTime)
 
+  def compare(a: LocalDate, y: Int, m: Int, d: Int)(f: (LocalDate, LocalDate) => Boolean) =
+    f(a, new LocalDate(y, m, d))
+
+  def show2016AddressMessage(address: Option[Address])(implicit messages: play.api.i18n.Messages): (Boolean, Option[String]) = {
```
[the remark this answers is not in the document]
Comment (richardjamesbaconHMRC) [reply app/util/DateTimeTools.scala]: show2016AddressMessage needs "compare" so I thought this was the correct place for it. Is this not the case?

File: test/controllers/AddressControllerSpec.scala [hunk 3]
```diff
@@ -152,7 +151,7 @@ class AddressControllerSpec extends BaseSpec  {
       override lazy val thisYearStr = "2015"
     }
 
-    "return OK if there is an entry in the cache to say the user previously visited the 'view address' page" in new LocalSetup {
+    "return OK if there is an entry in the cache to say the user previously visited the 'view address' or 'personal details' page" in new LocalSetup {
```
Comment (nicf82) [inline test/controllers/AddressControllerSpec.scala]: Is this descrition still correct?

**Pull request merged**